File: Assets/Fruit Cut Ninja Starter Kit/Scripts/FruitDispenser.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class FruitDispenser : MonoBehaviour
{
    
    public GameObject[] fruits;
    public GameObject bomb;
    public float y;
    public float powerScale;
    public bool pause = false;
    bool started = false;
    public float timer=2.5f;
    public int if_start_sended = 0;
    //判断是否是第一次接收到
    public bool is_first_start = false;

    public bool is_really_reieve;
    public bool is_really_reieve1;
    // Start is called before the first frame update
    void Start()
    {
       
    }

    // Update is called once per frame
    void FixedUpdate()
    {
        if (pause) return;
        timer -= Time.deltaTime;
        if(timer<=0&&!started)
        {
            timer = 0f;
            started = true;
        }
        if(started)
        {
            if (SharedSetting.LoadLevel == 0)
            {
                if (timer <= 0)
                {
                    FireUp();
                    timer = 1f;//水果出现的间隔
                }
            }
            else
            if (SharedSetting.LoadLevel == 1)
            {
                if (timer <= 0)
                {
                    FireUp();
                    timer = 2.0f;
                }
            }
            else
            if (SharedSetting.LoadLevel == 2)
            {
                if (timer <= 0)
                {
                    FireUp();
                    timer = 1.75f;
                }
            }
            else
            if (SharedSetting.LoadLevel == 3)
            {
                if (timer <= 0)
                {
                    FireUp();
                    timer = 1.5f;
                }
            }
        }
        //Spawn(false);
        //GameObject go = GameObject.Find("ins");

    }
    void FireUp()
    {
        if (pause) return;
        Spawn(false);
        if(SharedSetting.LoadLevel==2&&Random.Range(0,10)<2)
        {
            Spawn(false);

        }
        if (SharedSetting.LoadLevel == 3 && Random.Range(0, 10) < 4)
        {
            Spawn(false);

        }
        if (SharedSetting.LoadLevel == 1 && Random.Range(0, 100) < 10)
        {
            Spawn(true);

        }
        if (SharedSetting.LoadLevel == 2 && Random.Range(0, 100) < 20)
        {
            Spawn(true);

        }
        if (SharedSetting.LoadLevel == 3 && Random.Range(0, 100) < 30)
        {
            Spawn(true);

        }

    }

     
   
    void Spawn(bool isBomb)
    {
        int Done;
        
        //float x = Random.Range(1.15f, 1.15f);
        //z = Random.Range(2.17f, 2.17f);
        float x = 1.15f;
        y = 1.7f;
        GameObject ins;


        if_start_sended = GameObject.Find("Socket").GetComponent<connect>().is_start_sended;
        is_really_reieve = GameObject.Find("hammer bloody (1)").GetComponent<NewKnifeRotate2>().judge_down;
        is_really_reieve1 = GameObject.Find("hammer bloody").GetComponent<NewKnifeRotate1>().judge_down;
        Debug.Log("is_really_reieve" + is_really_reieve);

        ins = Instantiate(fruits[Random.Range(0, fruits.Length)], transform.position + new Vector3(x, y, 1.33f), Quaternion.Euler(0.0f, 2.0f, 0.0f)) as GameObject;
        is_first_start = true;
       

      

    }

    //void OnTriggerEnter(Collider other)
    //void OnTriggerEnter(Collider other)
    //{
    //    Destroy(other.gameObject);
        
    //}

  
}

File: Assets/Fruit Cut Ninja Starter Kit/Scripts/FruitDispenser.js
//fruit dispenser - dispense fruits with differend speed and time and gravitation

#pragma strict
var fruits : GameObject[];
var bomb : GameObject;

var z : float;
var sfx : AudioClip;
var pause : boolean = false;
var timer : float = 3.0;
private var started : boolean = false;
private var powerMod : float;

function Awake() {

if (SharedSettings.LoadLevel==1) {
	Random.seed = 356355;
	Physics.gravity.y = -2;
	powerMod = 1.0;
}
if (SharedSettings.LoadLevel==2) {
	Random.seed = 12411245;
	Physics.gravity.y = -3;
	powerMod = 0.825;
}
if (SharedSettings.LoadLevel==3) {
	Random.seed = 898979;
	Physics.gravity.y = -4;
	powerMod = 0.70;
}
if (SharedSettings.LoadLevel==4) {
	Random.seed = 64223459;
	Physics.gravity.y = -5;
	powerMod = 0.625; 
}

}

function Update() {

if (pause) return;

timer -= Time.deltaTime;

if (timer<=0.0 && !started) {
timer = 0.0;
started = true;
}

if (started) {

if (SharedSettings.LoadLevel==1) {
	if (timer<=0.0) {
		FireUp();
		timer = 2.0;
	}
}

if (SharedSettings.LoadLevel==2) {
	if (timer<=0.0) {
		FireUp();
		timer = 1.75;
	}
}

if (SharedSettings.LoadLevel==3) {
	if (timer<=0.0) {
		FireUp();
		timer = 1.5;
	}
}

if (SharedSettings.LoadLevel==4) {
	if (timer<=0.0) {
		FireUp();
		timer = 1.0;
	}
}

}

}

function Spawn(isbomb : boolean) {

	var x : float = Random.Range(-4.5,4.5);
	z += 1;
	if (z>=5.0) z = 0;
	var ins : GameObject;
	
	if (!isbomb) 
	ins = GameObject.Instantiate(fruits[Random.Range(0,fruits.length)],transform.position + Vector3(x,0,z),Random.rotation);
	else
	ins = GameObject.Instantiate(bomb,transform.position + Vector3(x,0,z),Random.rotation);

	var power = Random.Range(1.5,1.8) * -Physics.gravity.y * 1.5 * powerMod;

	var direction = Vector3(-x * 0.05 * Random.Range(0.3,0.8),1,0);
	direction.z = 0.0;

	ins.rigidbody.velocity =  direction * power;
	
	audio.PlayOneShot(sfx,1.0);
	ins.rigidbody.AddTorque(Random.onUnitSphere * 0.1,ForceMode.Impulse);

}

function FireUp () {

	if (pause) return;

	Spawn(false);
	
	if (SharedSettings.LoadLevel==2 && Random.Range(0,10)<2) {
		Spawn(false);
	}

	if (SharedSettings.LoadLevel==3 && Random.Range(0,10)<4) {
		Spawn(false);
	}

	if (SharedSettings.LoadLevel==4 && Random.Range(0,10)<4) {
		Spawn(false);
	}
	if (SharedSettings.LoadLevel==4 && Random.Range(0,10)<2) {
		Spawn(false);
	}

	//bomby
	if (SharedSettings.LoadLevel==2 && Random.Range(0,100)<10) {
		Spawn(true);
	}
	if (SharedSettings.LoadLevel==3 && Random.Range(0,100)<20) {
		Spawn(true);
	}
	if (SharedSettings.LoadLevel==4 && Random.Range(0,100)<30) {
		Spawn(true);
	}


}

    function OnTriggerEnter (other : Collider) {
        Destroy(other.gameObject);
    }